### ISSUE TITLE
properly install / configure prezto (aka sorin omz)

### ISSUE THREAD

**Issue by kylewest** [skwp/dotfiles#124]
## TL;DR
- Pick the best tool and make things just work.
- Sorin OMZ is the best OMZ but is missing a couple features.
- Adopt Sorin 100% (we're opinionated).
- Push a bunch of our zsh "stuff" to Sorin to simplify our config and so we're not duplicating effort.
## Full Issue

I'm late to the sorin omz party, but starting to check it out. After reviewing the [README](https://github.com/sorin-ionescu/oh-my-zsh) I realized we're not installing it properly (or at least not per the README). 

In addition to installing we need to do a better job detecting which OMZ we're using and act accordingly -- or set a default.

Installing per the README is going to cause a couple issues:
1. Sorin wants us to copy 4 files from his repo into the home directory. These files would not be under version control and changes to them would be lost (or have to be merged) when Sorin updates.
2. From what I can tell there is no way to customize Sorin OMZ without maintaining a fork. Hence, our `zsh` and `custom/zsh` folders would need to be fixed.
3. The project is [about to be renamed](https://github.com/sorin-ionescu/oh-my-zsh/issues/197). If we're still concerned about backwards compatibility we should wait until that happens to do any of this.
## Sorin vs. RobbyRussel vs. YADR

Some, and maybe a lot, of the Sorin stuff is incompatible with the legacy stuff. Eg. legacy has plugins, sorin has modules (#119).

Also, some of our stuff is likely duplicate of sorin built-ins. E.g. aliases (especially git), history-search, fasd, etc.
## Options

I like the "pick the best tool" and "just works" feel of our README. If the community is on board we should adopt Sorin 100% and contribute our various ZSH customizations to that project. We'll benefit from gaining the support of the sorin OMZ userbase and reduce our support burden on purely ZSH issues. Also, assuming sorin is more responsive to support requests we can quickly integrate our changes into his project.

Specifically ...
1. Package and submit our customization strategy. It seems to work well and we could hook into it to load YADR customizations.
2. Package and submit large customizations as new modules or enhancements to existing. For instance: we have a lot of vim specific "stuff" ... make a vim module. We have a ton of git aliases, enhance the [git module](https://github.com/sorin-ionescu/oh-my-zsh/issues/96). We do a good job loading rvm or rbenv ... add it as a module. Submit our themes.
3. Basically, move as much of our zsh "stuff" into the Sorin project.
## Related Issues
#97, #95

/cc @sorin-ionescu @skwp
